core/src/main/java/org/apache/iceberg/util/JsonUtil.java

```diff
@@ -189,6 +189,22 @@ public static String getStringOrNull(String property, JsonNode node) {
     return getString(property, node);
   }
 
+  public static String getDurationStringOrNull(String property, JsonNode node) {
+    String value = getStringOrNull(property, node);
+    if (value == null) {
+      return null;
+    }
+
+    try {
+      java.time.Duration.parse(value);
+    } catch (RuntimeException e) {
+      throw new IllegalArgumentException(
+          String.format("Cannot parse to a duration string value: %s: %s", property, value), e);
```

Comment:
```suggestion
          String.format("Cannot parse to a duration string value: %s: %s", property, node.get(property), e);
```
for consistency with other JsonUtil methods we should probably add the actual JSON node into the error msg